### ISSUE TITLE
[Feature Request] Add a way to create a RiiTag without Discord

### ISSUE THREAD

**Issue by osfanbuff63** [RiiConnect24/riitag#57]
## What do you want to be added?
I think that RiiTag shouldn't be limited to Discord accounts. In my case, I don't feel comfortable using Discord given how many times it has had security problems, therefore **potentially** allowing attackers to get a hold of your information.

## How could this be implemented?
I've thought of two potential options (with pros and cons listed):

- Have more SSO options.
    -  This would fix this issue, but may not be ideal since there are a lot of account types out there.
- Have a RiiTag account system
    -  For me, this is ideal, allowing you to create an account with your desired tag. However, auth servers are a problem in that regard.

I hope it's appreciated the amount of time I took to write this feature request. :)

**Comment by larsenv** [comment]
I don't see why you would avoid Discord just because it makes you feel uncomfortable for security issues (which isn't much of a problem), but the good news is that there's a major update planned for RiiTag that hasn't been deployed yet. I believe it will let you connect a Twitter account, but I'm not 100% sure.

**Comment by TheShadowEevee** [comment]
I second the above comment, I personally believe more SSO options are the way to go as storing account information would be a whole different world of legal concerns in storing emails, hashed passwords, etc.

I do also believe Twitter is planned for the upgrade, but I would be happy to look into other SSO options such as Steam and Google (This would cover the most general options most people will have) although I cannot promise anything there.

While I am personally unaware of security-specific problems with Discord myself, I personally agree with your request for more options in the future, so thank you for your feature request!

**Comment by KcrPL** [comment]
I also agree that Discord is currently the most secure way of authenticating.

**Comment by osfanbuff63** [comment]
I think Steam, Google, Twitter, and of course Discord, would cover the vast majority, as almost everyone I know would have one of those. The only other one I could think of that *could* be relevant is Microsoft (assuming the list above), but I don't think it's needed entirely.

**Comment by bendevnull** [comment]
The reason that Discord is used currently is because Discord is pretty much relied on for rendering parts of the tag; at least out of the box. However, Larsen does have a point with the new version of RiiTag coming out. It's been in the works for a while now, and while it has hit a small stall in development, we're working on getting it up and running again soon to hopefully deploy. I'll look into adding more SSO options, but we're also not going to be handling authentication on our end as that would be far to insecure and really just unneeded.

TL;DR: More SSO options will not be added to RiiTag as of now, but the next version of RiiTag will have more than Discord on release.
